### PR TITLE
Fix PushPoper animations

### PR DESCRIPTION
**Pull request by Iamsamwhoami**
I was debugging why the title of a pushed view controller just suddenly pops into it's final position without performing a transition as it should, found out that `setViewControllers(vcs, animated: animated)` was the culprit, so I guess something is flawed inside UIKit.

Not so sure about this fix, but it does fix the animations.

**Before:**

**After:**

### CODE REVIEW

File: Presentation/UINavigationController+Presenting.swift [hunk 1]
```diff
@@ -176,7 +176,14 @@ private extension UINavigationController {
                 setViewControllers(vcs, animated: animated)
             }
         } else {
-            setViewControllers(vcs, animated: animated)
+            // push and pop animations doesn't work for some reason with setViewControllers, so defaulting to pushViewController and popViewController
+            if vcs.count == viewControllers.count + 1 {
+                pushViewController(vcs.last!, animated: animated)
+            } else if vcs.count == viewControllers.count - 1 {
```
Comment (mansbernhardt) [inline Presentation/UINavigationController+Presenting.swift]: These two checks assumes that the rest of the view controllers in `vcs` and `viewControllers` are the same. I think we can make this check more correct with something like this instead?:

```swift
if vcs.elementsEqual(viewControllers.dropLast()) {
  pushViewController(vcs.last!, animated: animated)
} else if vcs.dropLast().elementsEqual(viewControllers.count) {
```

**Comment by mansbernhardt** [comment]
Great find and work around. Have you investigated if this is an issue just for certain iOS versions or for all support ones?

File: Presentation/UINavigationController+Presenting.swift (hunk 1 above)
Comment (Iamsamwhoami) [reply Presentation/UINavigationController+Presenting.swift]: @mansbernhardt oh, that's way better! I've updated the PR!

**Comment by Iamsamwhoami** [comment]
@mansbernhardt I checked in the simulator, worked as it should in iOS 10, but since iOS 11 it doesn't.

**Comment by mansbernhardt** [comment]
@mansbernhardt What is the status on this PR?

**Comment by Iamsamwhoami** [comment]
@mansbernhardt fine for merging by me (I don't have merge rights), though the tests are failing for some reason, doesn't seem related tho 🤔

**Comment by mansbernhardt** [comment]
@Iamsamwhoami Do the unit tests pass locally? I can see that they fail on the build server:

```
ExampleUITests.ExampleUITests
  testNavigationPresentationStyle, Lost connection to the application (pid 3093).
  ExampleUITests.swift:198
```

Not sure why, perhaps @nataliq can shed some light?

**Comment by nataliq** [comment]
Hey @Iamsamwhoami @mansbernhardt Sorry for the late response, I missed this mention here!
I restarted the build few times on the CI to check if it's flaky but it fails every time. After running the UI tests on the `StylesAndOptions` example app locally I see that the test is failing after these changes (there is a crash actually), so perhaps it is a good catch. If you run that app in the simulator the crash is reproducible every time and looks related to the change: `'Pushing the same view controller instance more than once is not supported`.

**Comment by Iamsamwhoami** [comment]
@nataliq ah, ok! I’ll check it out.

**Comment by Iamsamwhoami** [comment]
I will come back to this at some point when I can figure out why the test fails 🙈

**Pull request closed without merging**